torch/jit/__init__.py

```diff
@@ -1731,6 +1731,31 @@ def __getstate__(self):
                 "Mixed serialization of script and non-script modules is not supported. " +
                 "For purely script modules use my_script_module.save(<filename>) instead.")
 
+        # Python magic methods do lookups on an object's class type, instead of looking up
+        # the desugared attribute. In order to continue to expose the magic methods
+        # of builtin-containers (ModuleList, Sequential, ModuleDict) to python we
+        # define magic methods here as a shim to the correct attribute.
+        def forward_magic_method(self, method_name, *args, **kwargs):
+            self_method = getattr(self, method_name)
+            if self_method == getattr(RecursiveScriptModule, method_name):
```

Comment:
Won't this break `__dir__` calls for normal `ScriptModule`s?

Reply:
What are you worried about breaking ? Shouldn't the behavior inherit from `ModuleList`, which does the following:

```
        keys = super(ModuleList, self).__dir__()
        keys = [key for key in keys if not key.isdigit()]
        return keys
```

(and which this currently calls into)